utils/simulation/cryo_service.py

```diff
@@ -23,19 +50,196 @@ async def sequencer(self, instance, value):
             await self.mode.write(1)
             await self.mode_string.write("SEQUENCER")
 
+    @mode.putter
+    async def mode(self, instance, value):
+        if value == 1:
+            await self.trigger_heater_sequencer()
+
 
 class JTPVGroup(PVGroup):
+    def __init__(self, prefix, ll_group, heater_group):
+        super().__init__(prefix)
+        self.ll_group: LiquidLevelPVGroup = ll_group
+        self.heater_group: HeaterPVGroup = heater_group
+
     readback = pvproperty(name="ORBV", value=30.0)
-    ds_setpoint = pvproperty(name="SP_RQST", value=30.0)
+    ds_setpoint = pvproperty(name="SP_RQST", value=92.0)
     manual = pvproperty(name="MANUAL", value=0)
     auto = pvproperty(name="AUTO", value=0)
     mode = pvproperty(name="MODE", value=0)
     man_pos = pvproperty(name="MANPOS_RQST", value=40.0)
     mode_string: PvpropertyString = pvproperty(name="MODE_STRING", value="AUTO")
 
+    @property
+    def max_ll(self):
+        max_ll = 100
+        return max_ll
+
+    @property
+    def min_ll(self):
+        min_ll = 0
+        return min_ll
+
+    @property
+    def total_heat(self):
+        total_heat = self.heater_group.pdiss + self.heater_group.readback.value
+        return total_heat
+
+    @property
+    def stable_heat(self):
+        stable_heat = 48
+        return stable_heat
+
+    @property
+    def net_heat(self):
+        net_heat = self.total_heat - self.stable_heat
+        return net_heat
+
+    @property
+    def ll_delta(self):
+        dll_dt = -8.174374050765241e-05 * self.net_heat
+        dt = 1  # seconds
+        ll_delta = dll_dt * dt
+        return ll_delta
+
+    @property
+    def ll_diff(self):
+        current = self.ll_group.downstream.value
+        target = self.ds_setpoint.value
+        ll_diff = current - target
+        return ll_diff
+
+    def is_at_setpoint(self):
+        current = self.ll_group.downstream.value
+        target = self.ds_setpoint.value
+        return current == target
+
+    def is_at_max_fill(self):
+        current = self.ll_group.downstream.value
+        return current >= self.max_ll
+
+    def is_empty(self):
+        current = self.ll_group.downstream.value
+        return current <= self.min_ll
+
+    # logic behind direction determination
+    # Case 1: current > target
+    # ll_diff = positive number
+    # min(-ll_diff, +ll_delta) ==> returns -ll_diff
```

Comment:
I think there's some conflation happening here: ll_diff as you've written it is only valid when the JT in in auto, where we've arbitrarily decided that the fill rate is on the order of 0.2 (so it has nothing to do with the Q0 calibration slope). When the JT is in manual, the calibration slope determines how quickly the level is rising/falling but there is no target. So in reality the direction is really as simple as `sign(diff)` (but again, only in auto)